Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/decompile/actions/ShiftPointerAction.java

```diff
@@ -0,0 +1,208 @@
+						SourceType.USER_DEFINED);
+					if (useDataTypes) {
+						HighFunctionDBUtil.commitReturnToDatabase(hfunction,
+							SourceType.USER_DEFINED);
+					}
+				}
+				catch (DuplicateNameException e) {
+					throw new AssertException("Unexpected exception", e);
+				}
+				catch (InvalidInputException e) {
+					Msg.showError(this, null, "Parameter Commit Failed", e.getMessage());
+				}
+			}
+			HighFunctionDBUtil.updateDBVariable(highSymbol, null, dt, SourceType.USER_DEFINED);
+			successfulMod = true;
+		}
+		catch (DuplicateNameException e) {
+			throw new AssertException("Unexpected exception", e);
+		}
+		catch (InvalidInputException e) {
+			Msg.showError(this, tool.getToolFrame(), "Shift Failed",
+				"Failed to shift pointer '" + highSymbol.getName() + "': " + e.getMessage());
+		}
+		finally {
+			program.endTransaction(transaction, successfulMod);
+		}
+	}
+
+	/**
+	 * Retype a function return value
+	 * @param program the program
+	 * @param highFunction the function to retype
+	 * @param dt the new datatype
+	 * @param tool the PluginTool
+	 */
+	private void retypeReturn(Program program, HighFunction highFunction, DataType dt, PluginTool tool) { 
+		boolean commitRequired = checkFullCommit(null, highFunction);
+		if (commitRequired) {
+			int resp = OptionDialog.showOptionDialog(tool.getToolFrame(),
+				"Parameter Commit Required",
+				"Shifting the pointer requires all other parameters to be committed!\nContinue with shift?",
+				"Continue");
+			if (resp != OptionDialog.OPTION_ONE) {
+				return;
+			}
+		}
+		boolean successfulMod = false;
+		int transactionID = program.startTransaction("Shift Pointer");
+		try {
+			if (dt.getDataTypeManager() != program.getDataTypeManager()) {
+				dt = program.getDataTypeManager().resolve(dt, null);
+			}
+			if (commitRequired) {
+				try {
+					HighFunctionDBUtil.commitParamsToDatabase(highFunction, true,
+						SourceType.USER_DEFINED);
+				}
+				catch (DuplicateNameException e) {
+					throw new AssertException("Unexpected exception", e);
+				}
+				catch (InvalidInputException e) {
+					Msg.showError(this, null, "Parameter Commit Failed", e.getMessage());
+				}
+			}
+			highFunction.getFunction().setReturnType(dt, SourceType.USER_DEFINED);
+			successfulMod = true;
+		}
+		catch (InvalidInputException e) {
+			Msg.showError(this, tool.getToolFrame(), "Shift Failed",
+				"Failed to shift pointer '" + getName() + "': " + e.getMessage());
+		}
+		program.endTransaction(transactionID, successfulMod);
+	}
+	
+	@Override
+	protected boolean isEnabledForDecompilerContext(DecompilerActionContext context) {
+		Function function = context.getFunction();
+		if (function == null || function instanceof UndefinedFunction) {
+			return false;
+		}
+
+		ClangToken token = context.getTokenAtCursor();
+		if (token.Parent() instanceof ClangReturnType) {
+			if (((ClangReturnType) token.Parent()).getDataType() instanceof Pointer) return true;
```

Comment:
Ghidra's Java Style Guide in section 4.3 says to use braces where optional so this should be 
```suggestion
			if (((ClangReturnType) token.Parent()).getDataType() instanceof Pointer) {
				return true;
			}
```

same for lines 197, 204 and 205

Reply:
I tried to follow the coding style the rest of the codebase was written in, but I never found a Style Guide. OT: Could someone link it in the DevGuide?